### PR TITLE
[CXP-220] Add work location name and address to user profiles

### PR DESCRIPTION
**Pull request by johnallers**
## Summary

- Enriches user profiles with worker attributes: title, status, start/end date, work email, manager info, department, level, and employment type fields (`employment_type`, `employment_name`, `employment_label`)
- Enriches user profiles with name fields (given, family, middle, formatted, preferred given/family)
- Adds work address from SCIM user data, extracting `locality`, `region`, `country` as top-level profile fields
- Adds `--expand-work-locations` config toggle to fetch work locations via `GET /work-locations` and enrich user profiles with `work_location_name` and `work_location_address` (work location address takes precedence over user work address for top-level locality/region/country)
- Implements three-phase `List()`: work locations → workers → users, with locations and workers cached in the session store
- Adds configurable `--base-url` flag for mock server testing
- Moves `display_name` from `Name` to `User`/`WorkerUser` structs to match Rippling API docs
- Adds `WithUserLogin` trait to user resources
- Refactors `ListWorkers` to use shared `doGet` helper

## Test plan

- [x] `go build ./cmd/baton-rippling` passes
- [x] `go test ./...` passes — unit tests cover profile field extraction, address handling, work location enrichment, nil/empty edge cases
- [x] Manual: run with `--expand-work-locations`, verify `work_location_name` and `work_location_address` in user profiles
- [x] Manual: run without flag, verify no extra API calls and unchanged behavior
- [x] Manual: verify `employment_type`, `employment_name`, `employment_label` emit distinct values

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/connector/users_test.go [hunk 1]
```diff
@@ -138,11 +168,256 @@ func TestUserResource_NoEmails(t *testing.T) {
...
+	worker := &client.Worker{
+		ID:     "worker-11",
+		UserID: "user-11",
+		Status: "ACTIVE",
+	}
+
+	r, err := userResource(user, worker, nil)
+	assert.NoError(t, err)
+	assert.Equal(t, "Leo Park", r.DisplayName)
+}
```
[the remark this answers is not in the document]
Comment (johnallers) [reply pkg/connector/users_test.go]: this is fine for now

File: pkg/client/models.go [hunk 2]
```diff
@@ -28,7 +28,13 @@ type Photo struct {
...
 
 type Name struct {
-	DisplayName string `json:"display_name"`
+	Formatted           string `json:"formatted,omitempty"`
+	GivenName           string `json:"given_name,omitempty"`
+	MiddleName          string `json:"middle_name,omitempty"`
+	FamilyName          string `json:"family_name,omitempty"`
+	PreferredGivenName  string `json:"preferred_given_name,omitempty"`
+	PreferredFamilyName string `json:"preferred_family_name,omitempty"`
+	DisplayName         string `json:"display_name"`
```
Comment (agustin-conductor) [inline pkg/client/models.go]: I think display_name is not part of the [name type](https://developer.rippling.com/documentation/rest-api/reference/list-workers#:~:text=object-,The%20user%27s%20name.,-formatted)

**Review by agustin-conductor — APPROVED** [review]

File: pkg/client/endpoints.go [hunk 3]
```diff
@@ -9,110 +9,81 @@ import (
...
+			defer res.Body.Close()
 			logBody(ctx, res.Body)
 		}
-		return nil, &ratelimitData, fmt.Errorf("failed to list workers: %w", err)
+		return &ratelimitData, fmt.Errorf("request to %s failed: %w", baseURL, err)
 	}
-
 	defer res.Body.Close()
+
 	if res.StatusCode < 200 || res.StatusCode >= 300 {
```
Comment (btipling) [inline pkg/client/endpoints.go]: I think this is a redundant status code check, uhttp already does this https://github.com/ConductorOne/baton-sdk/blob/31c1b474d3cbe77195c419baa6743b35b709990f/pkg/uhttp/wrapper.go#L510

File: pkg/client/endpoints.go [hunk 4]
```diff
@@ -9,110 +9,81 @@ import (
...
-	if nextLink != "" {
-		url = nextLink
+func (c *Client) ListWorkers(ctx context.Context, nextLink string) (*WorkersResponse, *v2.RateLimitDescription, error) {
+	// Build the base URL with expand params so that doGet handles the rest.
+	// The expand param is only needed on the initial request; next_link URLs
+	// already include it.
+	baseURL := c.workersURL()
+	if expand := c.buildExpandParam(); expand != "" {
+		baseURL += "?expand=" + expand
 	}
```
Comment (btipling) [inline pkg/client/endpoints.go]: nite: I would use the url here

```go
 u, _ := url.Parse(c.workersURL())
  if expand := c.buildExpandParam(); expand != "" {
      q := u.Query()
      q.Set("expand", expand)
      u.RawQuery = q.Encode()
  }
  baseURL := u.String()
```

https://pkg.go.dev/net/url#URL.Query

**Review by btipling — APPROVED** [review]
minor nits

File: pkg/client/endpoints.go (hunk 3 above)
Comment (johnallers) [reply pkg/client/endpoints.go]: updated
